openedx/features/partners/tests/test_g2a_registration.py

```diff
@@ -0,0 +1,107 @@
+from ddt import ddt, data
+
+from django.core.urlresolvers import reverse
+from django.contrib.auth.models import User
+
+from openedx.features.partners.tests.factories import PartnerFactory, OrganizationFactory
+from openedx.core.lib.api.test_utils import ApiTestCase
+from lms.djangoapps.onboarding.models import Organization
+
+
+@ddt
+class RegistrationViewTest(ApiTestCase):
+    """Tests for the registration"""
+
+    NAME = "bob james"
+    COUNTRY = "Pakistan"
+    ORGANIZATION = "arbisoft"
+    EMAIL = "bob@example.com"
+    USERNAME = "bob"
+    PASSWORD = "Test@12345"
+    TERMS_OF_SERVICE = True
+
+    def setUp(self):
+        super(RegistrationViewTest, self).setUp()
+        self.organization = OrganizationFactory(
+            label='arbisoft'
+        )
+        self.partner = PartnerFactory.create(slug="give2asia", label="arbisoft")
+        self.url = reverse("partner_register", args=["give2asia"])
+
+    def test_put_not_allowed(self):
+        response = self.client.put(self.url)
+        self.assertHttpMethodNotAllowed(response)
+
+    def test_delete_not_allowed(self):
+        response = self.client.delete(self.url)
+        self.assertHttpMethodNotAllowed(response)
+
+    def test_register(self):
+        # Create a new partner user
+        response = self.client.post(self.url, {
+            "name": self.NAME,
+            "organization_name": self.organization.label,
+            "country": self.COUNTRY,
+            "email": self.EMAIL,
+            "username": self.USERNAME,
+            "password": self.PASSWORD,
+            'terms_of_service': self.TERMS_OF_SERVICE
+        })
+        self.assertHttpOK(response)
+        user = User.objects.get(username=self.USERNAME)
+        organization = Organization.objects.filter(label__iexact=self.organization.label).first()
+
+        self.assertEqual(self.USERNAME, user.username)
```

Comment:
Let's update logic here, if user is not found in DB, you will get exception here, our good is to have failed test instead

Reply:
done